python/try_parser.py

```diff
@@ -0,0 +1,163 @@
+    elif s == "mac-wpt-2013":
+        return JobConfig("MacOS WPT legacy-layout", OS.MACOS, wpt_layout=Layout.layout2013)
+    elif s == "mac-wpt-2020":
+        return JobConfig("MacOS WPT layout-2020", OS.MACOS, wpt_layout=Layout.layout2020)
+    elif s == "webgpu":
+        return JobConfig("WebGPU CTS", OS.LINUX,
+                         wpt_layout=Layout.layout2020,  # reftests are mode for new layout
+                         wpt_tests_to_run="_webgpu",  # run only webgpu cts
+                         profile="production",  # WebGPU works to slow with debug assert
+                         unit_tests=False)  # production profile does not work with unit-tests
+    else:
+        return None
+
+
+class Encoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, (Config, JobConfig)):
+            return o.__dict__
+        if isinstance(o, Layout):
+            return o.to_string()
+        return json.JSONEncoder.default(self, o)
+
+
+class Config(object):
+    def __init__(self, s: str | None = None):
+        self.fail_fast: bool = False
+        self.matrix: list[JobConfig] = list()
+        if s is not None:
+            self.parse(s)
+
+    def parse(self, s: str):
+        s = s.lower().strip()
+
+        if not s:
+            s = "full"
+
+        words: list[str] = s.split(" ")
+
+        for m in words:
+            # handle keywords
+            if m in ["fail-fast", "failfast", "fail_fast"]:
+                self.fail_fast = True
+                continue  # skip over keyword
+            if m == "full":
+                words.extend(["linux", "macos", "windows"])
+                continue  # skip over keyword
+
+            # handle presets
+            p = preset(m)
+            if p is None:
+                print(f"Ignoring wrong preset {m}")
+            else:
+                self.matrix.append(p)
+
+    def toJSON(self) -> str:
+        return json.dumps(self, cls=Encoder)
+
+
+def main():
+    conf = Config(" ".join(sys.argv[1:]))
+    print(conf.toJSON())
+
+
+if __name__ == "__main__":
+    main()
+
+
+import unittest # noqa
+import logging # noqa
+
+
+class TestParser(unittest.TestCase):
+    def test_string(self):
+        self.assertEqual(Config("linux fail-fast").toJSON(),
+                         '{"fail_fast": true, "matrix": [\
+{"name": "Linux", "os": "linux", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}]}')
+
+    def test_empty(self):
+        self.assertEqual(Config("").toJSON(),
+                         '{"fail_fast": false, "matrix": [\
+{"name": "Linux", "os": "linux", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}, \
+{"name": "MacOS", "os": "macos", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}, \
+{"name": "Windows", "os": "windows", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}]}')
```

Comment:
It would be great to add a test that simulates two presets one after another, which is what the labels path triggers.